raphtory-graphql/src/model/schema/mod.rs

```diff
@@ -1,3 +1,41 @@
+use raphtory::{
+    db::graph::vertex::VertexView,
+    prelude::{GraphViewOps, VertexViewOps},
+};
+use std::collections::{HashMap, HashSet};
+
+pub(crate) mod edge_echema;
 pub(crate) mod graph_schema;
+pub(crate) mod layer_schema;
 pub(crate) mod node_schema;
 pub(crate) mod property_schema;
+
+const ENUM_BOUNDARY: usize = 20;
+
+fn get_vertex_type<G: GraphViewOps>(vertex: VertexView<G>) -> String {
+    let prop = vertex.properties().get("type");
+    prop.map(|prop| prop.to_string())
+        .unwrap_or_else(|| "NONE".to_string())
+}
+
+type SchemaAggregate = HashMap<String, HashSet<String>>;
+
+fn merge_schemas(s1: SchemaAggregate, s2: SchemaAggregate) -> SchemaAggregate {
+    let mut merged_map = s1.clone();
```

Comment:
could you not just mark the input argument as mut? you are taking ownership already